### PR TITLE
Add occupation to vehicles

### PR DESCRIPTION
**Pull request by Greenscreen23**
### PR Checklist

Please make sure to fulfil the following conditions before marking this PR ready for review:

-   [x] If this PR adds or changes features or fixes bugs, this has been added to the changelog
-   [ ] ~If this PR adds new actions or other ways to alter the state, [test scenarios](https://github.com/hpi-sam/digital-fuesim-manv-public-test-scenarios) have been added~

### CODE REVIEW

File: shared/src/models/utils/occupations/intermediate-occupation.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    @IsInt()
+    @Min(0)
+    readonly unoccupiedUntil: number;
+
+    /**
+     * @deprecated Use static `create` method instead.
+     */
+    constructor(unoccupiedAfter: number) {
+        this.unoccupiedUntil = unoccupiedAfter;
```
Comment (Nils1729) [inline shared/src/models/utils/occupations/intermediate-occupation.ts:18]: I sense some inconsistent naming here

File: shared/src/models/utils/occupations/occupation-helpers-mutable.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import type { Mutable } from '../../../utils';
+import { cloneDeepMutable } from '../../../utils';
+import type { ExerciseOccupation } from './exercise-occupation';
+import { NoOccupation } from './no-occupation';
+
+export function isUnoccupied(
+    occupied: Mutable<{ occupation: ExerciseOccupation }>,
```
Comment (Nils1729) [inline shared/src/models/utils/occupations/occupation-helpers-mutable.ts:7]: Please also add functions for the case when we do not have a `Mutable` at hand.

File: shared/src/models/utils/occupations/occupation-helpers-mutable.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        occupied.occupation.type === 'intermediateOccupation' &&
+        occupied.occupation.unoccupiedUntil < currentTime
+    ) {
+        occupied.occupation = cloneDeepMutable(NoOccupation.create());
+    }
+
+    return occupied.occupation.type === 'noOccupation';
+}
+
+export function isUnoccupiedOrIntermediaryOccupied(
```
Comment (Nils1729) [inline shared/src/models/utils/occupations/occupation-helpers-mutable.ts:20]: I have not found the adverb in any dictionary but something like `intermediarily` should get the point across.

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ and this project does **not** adhere to [Semantic Versioning](https://semver.org
 -   The trainer editor now offers pre-configured templates for different kinds of simulated regions.
     -   These templates come with different border colors.
 -   Simulated regions display their names on the map.
+-   Vehicles can now have a property to indicate their current occupation.
```
Comment (Nils1729) [inline CHANGELOG.md:15]:
```suggestion
-   Vehicles now have a property to indicate their current occupation.
```

**Review by Nils1729 — COMMENTED** [review]
Please also add a small test scenario with the altered action and state.

File: shared/src/models/utils/occupations/intermediate-occupation.ts (hunk 1 above)
Comment (Greenscreen23) [reply shared/src/models/utils/occupations/intermediate-occupation.ts]: 2b95c61fffc46093fbdc363bb2b0e79f5b12b967

File: shared/src/models/utils/occupations/occupation-helpers-mutable.ts (hunk 2 above)
Comment (Greenscreen23) [reply shared/src/models/utils/occupations/occupation-helpers-mutable.ts:7]: 2b95c61fffc46093fbdc363bb2b0e79f5b12b967

File: shared/src/models/utils/occupations/occupation-helpers-mutable.ts (hunk 3 above)
Comment (Greenscreen23) [reply shared/src/models/utils/occupations/occupation-helpers-mutable.ts]: 2b95c61fffc46093fbdc363bb2b0e79f5b12b967

File: CHANGELOG.md (hunk 4 above)
Comment (Greenscreen23) [reply CHANGELOG.md]: 2b95c61fffc46093fbdc363bb2b0e79f5b12b967

**Comment by Greenscreen23** [comment]
Merge [digital-fuesim-manv-public-test-scenarios#1](https://github.com/hpi-sam/digital-fuesim-manv-public-test-scenarios/pull/1) when this gets merged

**Review by Nils1729 — APPROVED** [review]

**Pull request merged**